pkgs/development/haskell-modules/lib/default.nix

```diff
@@ -354,4 +354,32 @@ rec {
   # same package in the (recursive) dependencies of the package being
   # built. Will delay failures, if any, to compile time.
   allowInconsistentDependencies = compose.allowInconsistentDependencies;
+
+
+  # Collect all the cabal2nix derivers (i.e. nix file generated from a call
+  # to cabal2nix) used in a list of packages and save them as a link farm.
+  # You can use this function in order to keep references to these files so
+  # they won't be garbage collected.
+
+  # Examples of use:
+  #
+  # - If you perfectly know the list of packages (it will include dependencies):
+  #
+  # ```
+  # haskell.lib.collectAllCabal2nixDeriver [hPkg1 hPkg2]
+  # ```
+  #
+  # - Using the list derived from ghcWithPackages, it will only generate the
+  # deriver for the packages generated by callHackage, e.g. PyF here. aeson and
+  # generic-lens will be ignored.
+  #
+  # ```
+  # haskell.lib.collectAllCabal2nixDeriver ((haskellPackages.ghcWithPackages
+  #  (p: [ p.aeson p.generic-lens (p.callHackage "PyF" "0.11.0.0" { }) ])).paths)
+  # ```
+
+  collectAllCabal2nixDeriver = drvs: pkgs.linkFarmFromDrvs "all-cabal2nix-deriver"
+    (builtins.filter (x: x!= null)
+      (builtins.map (x: x.passthru.cabal2nixDeriver or null)
+        (lib.closePropagation drvs)));
```

Comment:
I have trouble understanding what‘s going on here. Can you maybe add an example what to do with this? i.e. where would I add this to my system config or shell.nix and why would I want this?

Reply:
@guibou Are you still interested in working on this?  I think responding here is the only thing preventing this PR from being merged in.

Reply:
I'm still interested.

Actually, we can remove this function. It is unclear and chances are that user interested by this feature already know how to write that by themself in a way suitable for their project.

Are you ok with the removal?

Reply:
Yeah, sure, we can always add it later.

Reply:
Are you targeting 21.11 still? In that case we need to backport as well (which still works), as branch-off happened.

Reply:
I removed that part of the changes, so this is no longer relevant. Only the initial fix remains. Tell me what you think about it.